### DISCUSSION TITLE
Running a Query of firebird 2.5 database the results with ç appear � instead.

### DISCUSSION THREAD

**Discussion by dyazbek21** [dbeaver/dbeaver#22653] [General]
I'm trying to visualize the results but instead ç appears � or #, but if I open the same database on IBExpert, appears ç
Both are UTF8 codification
Someone can help me?
![Screenshot_1](https://github.com/dbeaver/dbeaver/assets/86021741/32062897-7af9-4af0-93c0-0a5f34624dc7)

**Comment by mkgrgis** [comment]
Could you inspect `SELECT * FROM RDB$DATABASE` especially `SELECT RDB$CHARACTER_SET_NAME FROM RDB$DATABASE`?

**Reply by dyazbek21** [reply]
The results from select:
![Screenshot_2](https://github.com/dbeaver/dbeaver/assets/86021741/d781cfff-613f-451e-9d7a-d7e526fb20b2)
![Screenshot_3](https://github.com/dbeaver/dbeaver/assets/86021741/b990bd04-83a8-4087-b818-c8258f13cdb3)

**Reply by mkgrgis** [reply]
Thanks, @dyazbek21 ! `NONE` is very ugly character set for Firebird. You will get this and other problems in all of ISO:SQL clients such as Firebird JDBC, https://github.com/ibarwick/firebird_fdw , `isql-fb`, [`fbsql`](https://github.com/ibarwick/fbsql). DBeaver is derivative of Firebird JDBC in this case. If you have not tell to your Firebird database (fdb) about used encoding you have no chances for correct ordering or correct displaying of text data from your database. `NONE` guarantees only well sorted and well displayed ASCII subset, no other. Unfortunately `NONE` character set was typical for Firebird databases from USA and Russia in opposition to most of Firebird databases from Europe inspected by me.

**Comment by dyazbek21** [comment]
Thank you! Appreciate your help!

Em sex., 2 de fev. de 2024 13:48, Михаил ***@***.***>
escreveu:
